x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/policy_template_form.tsx

```diff
@@ -73,6 +79,114 @@ interface IntegrationInfoFieldsProps {
   onChange(field: string, value: string): void;
 }
 
+type AwsAccountType = 'single_account' | 'organization_account';
+
+const getAwsAccountTypeOptions = (isAwsOrgDisabled: boolean) => {
+  const awsAccountTypeOptions: CspRadioGroupProps['options'] = [
+    {
+      id: 'single_account',
+      label: i18n.translate('xpack.csp.fleetIntegration.awsAccountType.singleAccountLabel', {
+        defaultMessage: 'Single Account',
+      }),
+    },
+    {
+      id: 'organization_account',
+      label: i18n.translate('xpack.csp.fleetIntegration.awsAccountType.awsOrganizationLabel', {
+        defaultMessage: 'AWS Organization',
+      }),
+      disabled: isAwsOrgDisabled,
+      tooltip: isAwsOrgDisabled
+        ? i18n.translate(
+            'xpack.csp.fleetIntegration.awsAccountType.awsOrganizationDisabledTooltip',
+            { defaultMessage: 'Supported from integration version 1.5.0 and above' }
+          )
+        : undefined,
+    },
+  ];
+
+  return awsAccountTypeOptions;
+};
+
+const getAwsAccountType = (
+  input: Extract<NewPackagePolicyPostureInput, { type: 'cloudbeat/cis_aws' }>
+): AwsAccountType | undefined => input.streams[0].vars?.['aws.account_type']?.value;
+
+const AWS_ORG_MINIMUM_PACKAGE_VERSION = '1.5.0';
+
+const AwsAccountTypeSelect = ({
+  input,
+  newPolicy,
+  updatePolicy,
+  packageInfo,
+}: {
+  input: Extract<NewPackagePolicyPostureInput, { type: 'cloudbeat/cis_aws' }>;
+  newPolicy: NewPackagePolicy;
+  updatePolicy: (updatedPolicy: NewPackagePolicy) => void;
+  packageInfo: PackageInfo;
+}) => {
+  // This will disable the aws org option for any version LOWER than 1.5.0
+  const isValidSemantic = semverValid(packageInfo.version);
+  const isAwsOrgDisabled = isValidSemantic
+    ? semverCompare(packageInfo.version, AWS_ORG_MINIMUM_PACKAGE_VERSION) < 0
+    : true;
+
+  const awsAccountTypeOptions = getAwsAccountTypeOptions(isAwsOrgDisabled);
+
+  useEffect(() => {
+    if (!getAwsAccountType(input)) {
+      updatePolicy(
+        getPosturePolicy(newPolicy, input.type, {
+          'aws.account_type': {
+            value: awsAccountTypeOptions[0].id,
+            type: 'text',
+          },
+        })
+      );
+    }
+    // we only wish to call this one time, once the options are ready
```

Comment:
I think to call the hook only one time on the first render you need to pass an empty array. Passing `awsAccountTypeOptions` will cause the hook to be called on each render as it's an array and get a new reference each time `getAwsAccountTypeOptions` is called. Or you can use `useMemo` to memoize the value of getAwsAccountTypeOptions and update it only when `isAwsOrgDisabled` changes, but this only needed if the value of `isAwsOrgDisabled` changes between renders

Reply:
In theory it should only change on version change which shouldn't happen between renders. in any case i removed it from dependencies completely now. couldn't reproduce the problem where `packageInfo.version` was `undefined` which caused me to add this dep in the first place.